torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -282,6 +287,21 @@ void Reducer::autograd_hook(VariableIndex index) {
     return;
   }
 
+  // Rebuild bucket only if 1) it is first time rebuilt 2) unused_parameters_
+  // is empty, currently it does not support when there is unused parameters
+  // 3) this backward pass needs to run all reduce.
+  // Here, we just dump tensors and their parameter indices into
+  // rebuilt_tensors_ and rebuilt_param_indices_, and then at the end of
+  // finalize_backward(), buckets will be rebuilt based on rebuilt_tensors_
+  // and rebuilt_param_indices_, and then will be broadcasted and intialized.
+  // Also we only need to dump tensors and parameter indcies of one replica.
+  if (!was_rebuilt_bucket_ && unused_parameters_.empty() &&
+      index.replica_index == 0) {
+    rebuilt_tensors_.push_back(
```

Comment:
call this `rebuilt_params_` to match the name of `rebuilt_param_indices_` below?